race_test.go

```diff
@@ -0,0 +1,60 @@
+package quicktest_test
+
+import (
+	"sync"
+	"sync/atomic"
+	"testing"
+
+	qt "github.com/frankban/quicktest"
+)
+
+func TestConcurrentMethods(t *testing.T) {
+	// This test is designed to be run with the race
+	// detector enabled. It checks that C methods
+	// are safe to call concurrently.
+
+	// N holds the number of iterations to run any given
+	// operation concurrently with the others.
+	N := 100
+
+	var x, y int32
+	c := qt.New(t)
+	var wg sync.WaitGroup
+	// start calls f in two goroutines, each
+	// running it N times.
+	start := func(f func()) {
+		repeat := func() {
+			defer wg.Done()
+			for i := 0; i < N; i++ {
+				f()
+			}
+		}
+		wg.Add(2)
+		go repeat()
+		go repeat()
+	}
+	start(func() {
+		c.Defer(func() {
+			atomic.AddInt32(&x, 1)
+		})
+		c.Defer(func() {
+			atomic.AddInt32(&y, 1)
+		})
+	})
+	start(func() {
+		c.Done()
+	})
+	start(func() {
+		c.SetFormat(func(v interface{}) string {
+			return "x"
+		})
+		// How can we exercise c.format for printing
+		// without failing the test?
```

Comment:
Using https://github.com/frankban/quicktest/blob/master/quicktest_test.go#L526 ?

Reply:
that's not thread-safe, but I added another minimal implementation for this test.